snappi_ixnetwork/trafficitem.py

```diff
@@ -191,9 +220,245 @@ class TrafficItem(CustomField):
+        imports = {}
+        imports["traffic"] = tr
+        self._importconfig(imports)
+        return ixn._connection._execute(url, payload)
+
+    def remove_ixn_traffic(self):
+        if len(self._api._ixnetwork.Traffic.TrafficItem.find()) > 0:
+            with Timer(self._api, "Remove Flows"):
+                start_states = [
+                    "txStopWatchExpected",
+                    "locked",
+                    "started",
+                    "startedWaitingForStats",
+                    "startedWaitingForStreams",
+                    "stoppedWaitingForStats",
+                ]
+                state = self._api._ixnetwork.Traffic.State
+                if state in start_states:
+                    self._api._ixnetwork.Traffic.StopStatelessTrafficBlocking()
+                url = "%s/traffic/trafficItem" % self._api._ixnetwork.href
+                self._api._request("DELETE", url)
+                self._api._ixnetwork.Traffic.TrafficItem.find().refresh()
+        self.traffic_index = 1
+
+    def create_traffic(self, config):
+        flows = config.flows
+        tr = {"xpath": "/traffic", "trafficItem": []}
+        ports = self.get_ports_xpath(config)
+        devices = self.get_device_xpath(config)
+        for i, f in enumerate(flows):
+            if f._properties.get("name") is None:
+                raise Exception("name shall not be null for flows")
+            if f._properties.get("tx_rx") is None:
+                msg = (
+                    "Please configure the flow endpoint"
+                    "for flow indexed at %s" % i
+                )
+                raise Exception(msg)
+            if f.tx_rx.choice is None:
+                msg = "Flow endpoint needs to be either port or device"
+                raise Exception(msg)
+            if f.tx_rx.choice == "port":
+                tr_type = "raw"
+                ep = getattr(f.tx_rx, "port")
+                tx_objs = ["%s/protocols" % ports.get(ep.tx_name)]
+                rx_objs = ["%s/protocols" % ports.get(ep.rx_name)]
+            else:
+                ep = getattr(f.tx_rx, "device")
+                tr_type = devices[ep.tx_names[0]]["type"]
+                tx_objs = [devices[n]["xpath"] for n in ep.tx_names]
+                rx_objs = [devices[n]["xpath"] for n in ep.rx_names]
+            tr_xpath = "/traffic/trafficItem[%d]" % self.traffic_index
+            tr["trafficItem"].append(
+                {"xpath": tr_xpath, "name": "%s" % f.name}
+            )
+
+            tr["trafficItem"][-1]["endpointSet"] = [
+                {"xpath": tr["trafficItem"][-1]["xpath"] + "/endpointSet[1]"}
+            ]
+            tr["trafficItem"][-1]["endpointSet"][0]["sources"] = [
+                o for o in tx_objs
+            ]
+            tr["trafficItem"][-1]["endpointSet"][0]["destinations"] = [
+                o for o in rx_objs
+            ]
+            tr["trafficItem"][-1]["trafficType"] = tr_type
+            if tr_type == "raw":
```

Comment:
In this new implementation we will add additional stack for device endpoint. 
eth, vlan, tcp = flow.packet.ethernet().vlan().tcp()
Then it will produced eth> IPv4 > vlan > tcp (device configure with IPv4)
This actually regress current implementation

Reply:
in the latest implementation, this is handled. but still we can validate once